### PR TITLE
Extract out 'is the html body element' CSSOM concept.

### PR DESCRIPTION
**Pull request by frewsxcv**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/10513)

### CODE REVIEW

File: components/script/dom/htmlbodyelement.rs [hunk 1]
```diff
@@ -47,6 +47,16 @@ impl HTMLBodyElement {
         let element = HTMLBodyElement::new_inherited(localName, prefix, document);
         Node::reflect_node(box element, document, HTMLBodyElementBinding::Wrap)
     }
+
+    /// https://drafts.csswg.org/cssom-view/#the-html-body-element
+    pub fn is_the_html_body_element(&self) -> bool {
+        let self_node = self.upcast::<Node>();
+        let root_elem = self.upcast::<Element>().get_root_element();
+        let root_node = root_elem.upcast::<Node>();
+        self_node.preceding_nodes(root_node).all(|n| !n.is::<HTMLBodyElement>()) &&
```
Comment (Ms2ger) [inline components/script/dom/htmlbodyelement.rs]: `preceding_siblings`, and swap around the two conditions, so the cheap/most-likely to fail check goes first.

Comment (frewsxcv) [inline components/script/dom/htmlbodyelement.rs]: Addressed in the latest force push.

File: components/script/dom/htmlbodyelement.rs [hunk 2]
```diff
@@ -47,6 +47,16 @@ impl HTMLBodyElement {
...
         Node::reflect_node(box element, document, HTMLBodyElementBinding::Wrap)
     }
+
+    /// https://drafts.csswg.org/cssom-view/#the-html-body-element
+    pub fn is_the_html_body_element(&self) -> bool {
+        let self_node = self.upcast::<Node>();
+        let root_elem = self.upcast::<Element>().root_element();
+        let root_node = root_elem.upcast::<Node>();
+        root_node.is_parent_of(self_node) &&
+            self_node.preceding_nodes(root_node).all(|n| !n.is::<HTMLBodyElement>())
```
Comment (Ms2ger) [inline components/script/dom/htmlbodyelement.rs]: siblings

Comment (frewsxcv) [inline components/script/dom/htmlbodyelement.rs]: Sorry, misread what you wrote the first time. Updated.

**Pull request merged**